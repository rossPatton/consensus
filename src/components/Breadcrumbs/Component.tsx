import React, { memo } from 'react';
import { Link } from 'react-router-dom';

export const BreadcrumbsComponent = memo((props: any) => (
  <ul className="fs6 fw600 mB2 lh1 lsNone fx fxWrap brdB1 mB3 pB3">
    {console.log('props => ', props)}
    {props.country && (
      <li className="mR1">
        {!props.region && props.country.name}
        {props.region && (
          <Link to={`/${props.country.code}`}>
            {props.country.name}
          </Link>
        )}
      </li>
    )}
    {props.region && (
      <li className="mHide mR1">/</li>
    )}
    {props.region && (
      <li className="mR1">
        {!props.city && props.country.name}
        {props.city && (
          <Link to={`/${props.country.code}/${props.region.code}`}>
            {props.region.name}
          </Link>
        )}
      </li>
    )}
    <li className="mHide mR1">/</li>
    <li className="mR1">
      <Link to="/us/ny/new-york-city">
        New York City
      </Link>
    </li>
  </ul>
));
