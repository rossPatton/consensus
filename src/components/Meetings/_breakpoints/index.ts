import loadable from '@loadable/component';

export const DesktopMeetings = loadable(() =>
  import(/* webpackChunkName: "Desktop Meetings Component" */'./Desktop'),
);

export const MobileMeetings = loadable(() =>
  import(/* webpackChunkName: "Mobile Meetings Component" */'./Mobile'),
);

