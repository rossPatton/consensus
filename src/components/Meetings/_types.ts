type tProps = {
  count?: number,
  meetings: ts.meeting[],
  // render alternative version
  horizontal?: boolean,
  // just because eslint complains about using role with non-ARIA strings
  sessionRole?: ts.role,
  showGroupName?: boolean,
  showPast?: boolean,
  showRSVPs?: boolean,
};

export type tStore = {
  session: ts.thunk<ts.session>,
};

export type tContainerProps = tProps & {
  deleteMeetingDispatch: (query: ts.idQuery) => ts.thunkPayload,
  session: ts.session,
  // realistically just 3 options:
  // admin 'drafts' vs plain 'meetings' vs user meeting 'rsvps'
  type?: ts.groupRouteParams['section'] | 'rsvps';
};

export type tComponentProps = ts.mediaContext & tProps & {
  deleteMeeting: (ev: React.MouseEvent, id: number) => void,
  // if user is an admin, they can edit meetings
  isEditable?: boolean,
};


