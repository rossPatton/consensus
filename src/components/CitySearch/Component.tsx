import _ from 'lodash';
import React, {memo} from 'react';

import stateCodeMap from '../../json/usa/stateCodeMap.json';
import {tComponentProps} from './_types';

export const CitySearchComponent = memo((props: tComponentProps) => {
  const onCityChange = (ev: React.ChangeEvent<HTMLSelectElement>) => {
    const city = _.find(props.cities, c => c.name === ev.currentTarget.value);
    if (city) {
      props.updateState({
        city: city.name,
        cityId: city.id,
        region: city.region,
        regionId: city.regionId,
      });
    }
  };

  const onRegionChange = (ev: React.ChangeEvent<HTMLSelectElement>) => {
    const region = ev.currentTarget.value;
    if (region !== props.region) {
      props.updateState({
        city: '',
        cityId: 0,
        region,
        regionId: 0,
      });
    }
  };

  return (
    <>
      {props.showRegionField && (
        <>
          <h2 className="fs5 mB1 lh1">
            Pick a different state
          </h2>
          <label htmlFor="stateSelect">
            <select
              name="type"
              id="stateSelect"
              className="mB4 row bgWhite"
              value={props.region}
              onBlur={onRegionChange}
              onChange={onRegionChange}>
              {Object.keys(stateCodeMap).map(state => (
                <option key={state} value={state}>
                  {state}
                </option>
              ))}
            </select>
          </label>
        </>
      )}
      <h2 className="fs5 mB1 lh1">
        City in <span className="dInBl mR2">
          {(!props.showRegionField && props.geo.region) && props.geo.region}
          {(props.showRegionField && props.region) && props.region}
        </span>
        {!props.showRegionField && (
          <button
            type="button"
            onClick={props.toggleRegionField}>
            Not the right state?
          </button>
        )}
      </h2>
      {props.label && (
        <p className="fs5 copyBlack mB1">
          {props.label}
        </p>
      )}
      <div className="mB4">
        {props.cityId === 0 && (
          <label htmlFor="citySelect">
            <select
              name="type"
              id="citySelect"
              className="row bgWhite"
              value={props.city}
              onBlur={onCityChange}
              onChange={onCityChange}>
              {props.cities.map((city: tCity) => (
                <option key={city.id} value={city.name}>
                  {city.name}, {city.region}
                </option>
              ))}
            </select>
          </label>
        )}
        {props.cityId > 0 && (
          <div className="brdA1 p3 black br8 dInBl">
            <b>{props.city}</b>, <span className="dInBl mR3">{props.region}</span>
            <button
              type="button"
              onClick={() => {
                props.updateState({
                  city: '',
                  cityId: 0,
                });
              }}>
              Not the right city?
            </button>
          </div>
        )}
      </div>
    </>
  );
});
