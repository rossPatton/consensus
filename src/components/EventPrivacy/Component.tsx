import React, {memo} from 'react';

import {tProps} from './_types';

export const EventPrivacyComponent = memo((props: tProps) => (
  <small className="bgYellowLite br8 p1 pL2 pR2 mR2">
    {props.isPrivate && (
      <>
        <span
          role="img"
          className="mR1"
          aria-label="Lock Emoji">
          ðŸ”’
        </span>
        Private Event
      </>
    )}
    {!props.isPrivate && (
      <>
        <span
          role="img"
          className="mR1"
          aria-label="Tada Emoji">
          ðŸŽ‰
        </span>
        Public Event
      </>
    )}
  </small>
));
