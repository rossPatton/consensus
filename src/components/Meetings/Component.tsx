import _ from 'lodash';
import React, {memo} from 'react';

import {DesktopEvents, MobileEvents} from './_breakpoints';
import {tComponentProps} from './_types';

export const EventsComponent = memo((props: tComponentProps) => {
  if (props.isDesktop) {
    return (
      <DesktopEvents
        {...props}
      />
    );
  }

  return (
    <MobileEvents
      {...props}
    />
  );
});
