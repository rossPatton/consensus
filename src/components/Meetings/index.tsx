import _ from 'lodash';
import React, {PureComponent} from 'react';
import {connect} from 'react-redux';

import {Paginate} from '~app/containers';
import {MediaContext} from '~app/context';
import {deleteEvent} from '~app/redux';
import {tContainerProps, tStore} from './_types';
import {MeetingsComponent} from './Component';

class MeetingsContainer extends PureComponent<tContainerProps> {
  static contextType = MediaContext;
  static defaultProps = {
    count: 4,
  };

  deleteEvent = (ev: React.MouseEvent, id: number) => {
    ev.preventDefault();
    this.props.deleteEventDispatch({id});
  }

  render() {
    const {
      count,
      meetings = [],
      sessionRole,
      showOrgName,
      showRSVPs,
      type = 'meetings',
    } = this.props;

    const {isMobile, isDesktop} = this.context;
    const isEditable = sessionRole === 'admin' || sessionRole === 'facilitator';

    if (meetings.length === 0) {
      return (
        <h2 className="fs4 p-3 text-center">
          No {type === 'meetings' && 'upcoming meetings!'} {type === 'drafts' && 'drafts'} {type === 'rsvps' && 'upcoming RSVPs!'}
        </h2>
      );
    }

    return (
      <Paginate
        count={count}
        items={meetings}
        render={(meetingsToRender: ts.meeting[]) => (
          <MeetingsComponent
            deleteEvent={this.deleteEvent}
            meetings={meetingsToRender}
            horizontal={this.props.horizontal}
            isEditable={isEditable}
            isMobile={isMobile}
            isDesktop={isDesktop}
            sessionRole={sessionRole}
            showOrgName={showOrgName}
            showRSVPs={showRSVPs}
          />
        )}
      />
    );
  }
}

const mapStateToProps = (store: tStore) => ({
  session: store.session.data,
});

const mapDispatchToProps = (dispatch: Function) => ({
  deleteEventDispatch: (query: ts.idQuery) => dispatch(deleteEvent(query)),
});

const Meetings = connect(mapStateToProps, mapDispatchToProps)(MeetingsContainer);
export default Meetings;
