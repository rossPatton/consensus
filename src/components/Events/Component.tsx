import cx from 'classnames';
import dayJS from 'dayjs';
import _ from 'lodash';
import qs from 'query-string';
import React, {memo} from 'react';
import {Link} from 'react-router-dom';

import {ExternalLink, PlaceholderImage, RSVP} from '../../components';
import {objToQueryString} from '../../utils';
import {tComponentProps} from './_types';

export const EventsComponent = memo((props: tComponentProps) => (
  <ul
    className={cx({
      'fx fxdRow taL': props.horizontal,
      'jcBetween': props.horizontal && props.events.length === 4,
    })}>
    {props.events.map((ev, i) => {
      const isPastEvent = dayJS(ev.date).isBefore(dayJS());

      return (
        <li
          key={ev.id}
          className={cx({
            'mB2': !props.horizontal,
            'col row': props.horizontal,
            'mR4': props.horizontal && i !== props.events.length - 1,
          })}>
          <div
            className={cx({
              'fx aiCtr': !props.horizontal,
              'p3 hvrBgGrey1 br8': props.isEditable,
            })}>
            <div
              className={cx({
                'mB2 bgWhite': props.horizontal,
                'mR3 bgGrey2': !props.horizontal,
              })}>
              <Link
                className="noUnderline"
                to={ev.isDraft
                  ? `/org/${ev.orgId}/planMeeting?${qs.stringify(ev)}`
                  : `/event/${ev.id}`}>
                <PlaceholderImage
                  height={100}
                  seed={ev.id}
                  width={200}
                />
              </Link>
            </div>
            <div>
              <div
                className={cx({
                  'fx aiCtr mB1 fs7 fw600 lh1': true,
                  o5: isPastEvent,
                })}>
                <time className="mR1" dateTime={ev.date}>
                  {dayJS(ev.date).format('MMM DD | h:mmA')}
                </time>
                {!props.horizontal && (
                  <>
                    <span className="mR1">@</span>
                    {ev.locationLink && (
                      <ExternalLink
                        noFollow
                        className="mR1"
                        to={ev.locationLink}>
                        {ev.location}
                      </ExternalLink>
                    )}
                  </>
                )}
                {!ev.locationLink && ev.location}
              </div>
              <h3
                className={cx({
                  'fx aiCtr ttCap': true,
                  mB1: props.isEditable,
                  fs4: props.horizontal,
                })}>
                <Link
                  className="noUnderline"
                  to={ev.isDraft
                    ? `/org/${ev.orgId}/planMeeting?${qs.stringify(ev)}`
                    : `/event/${ev.id}`}>
                  {ev.title}
                </Link>
              </h3>
              {props.showRSVPs && (
                <div
                  className={cx({
                    'fx aiCtr fs6 lh1': true,
                    hide: props.horizontal,
                    o5: isPastEvent,
                  })}>
                  {!ev.isDraft && <RSVP event={ev} />}
                  {ev.isDraft && 'This event is not published yet. Click to edit.'}
                </div>
              )}
              {props.showOrgName
                && (
                  <Link
                    to={`/org/${ev.orgId}`}
                    className={cx({
                      'fw600 fs6 lh1': true,
                      o5: isPastEvent,
                    })}>
                    {ev.orgName}
                  </Link>
                )}
              {!isPastEvent
              && props.isEditable
              && (
                <div className="fx aiCtr mB1">
                  <div className="bgGrey4 br4 fs7 lh1 mR2 p1 pL2 pR2 white">
                    {ev.isDraft ? 'Draft' : 'Published'}
                  </div>
                  <div className="fx aiCtr mR1">
                    <Link
                      to={`/org/${ev.orgId}/planMeeting?${objToQueryString(ev)}`}
                      className="btn fs7 fw600 hvrBgGrey1 lh1 noUnderline p1 pL2 pR2">
                      <span
                        role="img"
                        aria-label="Hand with Pen Emoji">
                        ✍️
                      </span>
                      Edit
                    </Link>
                  </div>
                  <button
                    onClick={e => props.deleteEvent(e, ev.id)}
                    className="aiCtr fs7 fw600 fx hvrBgGrey1 lh1">
                    <span
                      role="img"
                      className="mR1"
                      aria-label="Big X Emoji">
                      ✖️
                    </span>
                    Delete
                  </button>
                </div>
              )}
            </div>
          </div>
        </li>
      );
    })}
  </ul>
));
