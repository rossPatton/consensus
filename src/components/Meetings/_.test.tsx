import React from 'react';
import { Provider } from 'react-redux';
import render from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import {testMeeting1, testMeeting2} from '~app/constants/jest';
import MatchMediaProvider from '~app/context/MatchMediaProvider';

import Meetings from '.';
import {MeetingsComponent} from './Component';

const mockStore = configureStore();

describe('components/Meetings', () => {
  it('renders mobile container without crashing', () => {
    const store = mockStore({
      session: {isAuthenticated: false},
    });

    const component = render.create((
      <Provider store={store}>
        <MatchMediaProvider isMobile>
          <Meetings meetings={[testMeeting1, testMeeting2]} />
        </MatchMediaProvider>
      </Provider>
    ));

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('desktop container with empty meetings array', () => {
    const store = mockStore({
      session: {isAuthenticated: false},
    });

    const component = render.create((
      <Provider store={store}>
        <MatchMediaProvider isDesktop>
          <Meetings meetings={[]} />
        </MatchMediaProvider>
      </Provider>
    ));

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders desktop component without crashing', () => {
    const component = render.create((
      <MeetingsComponent
        isDesktop
        deleteMeeting={jest.fn()}
        isMobile={false}
        meetingsToRender={[testMeeting1]}
        pastMeetings={[]}
        upcomingMeetings={[]}
        togglePast={jest.fn()}
      />
    ));

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders mobile component without crashing', () => {
    const component = render.create((
      <MeetingsComponent
        isMobile
        deleteMeeting={jest.fn()}
        isDesktop={false}
        meetingsToRender={[]}
        pastMeetings={[]}
        upcomingMeetings={[]}
        togglePast={jest.fn()}
      />
    ));

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
