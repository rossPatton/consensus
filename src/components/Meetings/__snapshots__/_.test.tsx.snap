// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Meetings desktop container with empty meetings array 1`] = `
<h2
  className="fs4 p-3 text-center"
>
  No 
  meetings!
   
   
</h2>
`;

exports[`components/Meetings renders desktop component without crashing 1`] = `null`;

exports[`components/Meetings renders mobile component without crashing 1`] = `null`;

exports[`components/Meetings renders mobile container without crashing 1`] = `null`;
